### ISSUE TITLE
Logistic Regression

### ISSUE THREAD

**Issue by Val11011** [eriklindernoren/ML-From-Scratch#124]
In class LogisticRegression, this line
```Python
self.param -= self.learning_rate * -(y - y_pred).dot(X)
```
indicates that the gradient is calculated as -(y-y_pred).dot(X).
Apparently, a factor of `1 / n_samples` is missed here.

The interesting part is that I found there is an issue about the similar problem in Regression.py.
It seems that the issue has been fixed, but only for Regression.py, not LogisticRegression.py

**Comment by Raheel-glicth** [comment]
hi @Val11011  is this still  problem or someone fixed that 
and if not i think to fix that
